### PR TITLE
Fix various documentation typos and formatting

### PR DESCRIPTION
**Pull request by bcc32**
Just a drive-by fix of some typos and dead links that I stumbled across.  No guarantee of exhaustiveness :)

### CODE REVIEW

File: lib-driver/caqti_driver_mariadb.mli [hunk 1]
```diff
@@ -12,14 +12,16 @@
  *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this library.  If not, see <http://www.gnu.org/licenses/>.
- *)
+*)
```
Comment (paurkedal) [inline lib-driver/caqti_driver_mariadb.mli:15]: Please don't remove the indentation of end-of-comment tokens.

File: lib/caqti_connection_sig.mli [hunk 2]
```diff
@@ -178,10 +178,10 @@ module type Convenience = sig
...
     ('a, 'b, [< `Zero | `One | `Many]) Caqti_request.t -> 'a ->
     ('b list, [> Caqti_error.call_or_retrieve] as 'e) result future
-  (** [rev_collect_list request param] performs a {!call} on [request],
-      extracting the result as a reversed list.  This is more efficient than
-      {!find_list} and fits well with a subsequent {!List.rev_map}, though it
-      may not matter much in practise. *)
+    (** [rev_collect_list request param] performs a {!call} on [request],
+        extracting the result as a reversed list.  This is more efficient than
+        {!collect_list} and fits well with a subsequent {!List.rev_map}, though it
+        may not matter much in practise. *)
```
Comment (paurkedal) [inline lib/caqti_connection_sig.mli:184]: Please don't add indentation here.

File: lib/caqti_connection_sig.mli [hunk 3]
```diff
@@ -195,9 +195,9 @@ module type Populate = sig
     columns: string list ->
     'a Caqti_type.t -> ('a, 'err) stream ->
     (unit, [> Caqti_error.call_or_retrieve | `Congested of 'err]) result future
-  (** [populate table columns row_type seq] inputs the contents of [seq] into
-      the database in whatever manner is most efficient as decided by the
-      driver. *)
+    (** [populate table columns row_type seq] inputs the contents of [seq] into
+        the database in whatever manner is most efficient as decided by the
+        driver. *)
```
Comment (paurkedal) [inline lib/caqti_connection_sig.mli:200]: Same.

File: lib/caqti_request.mli [hunk 4]
```diff
@@ -135,16 +134,16 @@ val create_p :
...
-      [env driver_info key] shall provide the value to substitute for a
-      reference to [key] in the preliminary query string, or raise [Not_found]
-      to indicate the reference to [key] is invalid.  [Not_found] will be
-      re-raised as [Invalid_argument] with additional information to help locate
-      the bug. *)
+    [env driver_info key] shall provide the value to substitute for a
+    reference to [key] in the preliminary query string, or raise [Not_found]
+    to indicate the reference to [key] is invalid.  [Not_found] will be
+    re-raised as [Invalid_argument] with additional information to help locate
+    the bug. *)
```
Comment (paurkedal) [inline lib/caqti_request.mli:146]: Please don't remove indentation here.

File: lib-driver/caqti_driver_mariadb.mli [hunk 5]
```diff
@@ -12,14 +12,16 @@
...
 
 (** Caqti driver for MariaDB (bindings).
 
-    This driver is implement in terms of the mariadb OPAM package.
+    This driver is implemented in terms of the mariadb OPAM package.
```
Comment (paurkedal) [inline lib-driver/caqti_driver_mariadb.mli:19]: Thanks for this and the other type fixes!

File: lib-driver/caqti_driver_mariadb.mli [hunk 6]
```diff
@@ -12,14 +12,16 @@
...
     It handles URIs of the form
 
-      {[mariadb://<user>:<password>@<host>:<port>/?socket=<socket>]}
+    {[
+      mariadb://<user>:<password>@<host>:<port>/?socket=<socket>
+    ]}
```
Comment (paurkedal) [inline lib-driver/caqti_driver_mariadb.mli:24]: Yes, maybe that looks better in the code, but I suggest also removing empty lines above and below.

File: lib/caqti_request.mli [hunk 7]
```diff
@@ -198,18 +197,18 @@ val pp_with_param :
...
-module Dynparam = struct
-  type t = Pack : 'a Caqti_type.t * 'a -> t
-  let empty = Pack (Caqti_type.unit, ())
-  let add t x (Pack (t', x')) = Pack (Caqti_type.tup2 t' t, (x', x))
-end
+      module Dynparam = struct
+        type t = Pack : 'a Caqti_type.t * 'a -> t
+        let empty = Pack (Caqti_type.unit, ())
+        let add t x (Pack (t', x')) = Pack (Caqti_type.tup2 t' t, (x', x))
+      end
```
Comment (paurkedal) [inline lib/caqti_request.mli:204]: IIRC the extra indentation showed up in the generated docs in old times, but it looks good now and definitely better to look at in the source code.

**Review by paurkedal — CHANGES_REQUESTED** [review]

File: lib-driver/caqti_driver_mariadb.mli (hunk 1 above)
Comment (bcc32) [reply lib-driver/caqti_driver_mariadb.mli:15]: Ah, sorry about that.  I just accepted whatever ocp-indent gave me in this case.

File: lib-driver/caqti_driver_mariadb.mli (hunk 6 above)
Comment (bcc32) [reply lib-driver/caqti_driver_mariadb.mli:23]: Fixed for all of the driver.mli files

File: lib/caqti_request.mli (hunk 4 above)
Comment (bcc32) [reply lib/caqti_request.mli]: Thanks.  I believe I have removed all of the unintended whitespace changes.

**Review by paurkedal — APPROVED** [review]
Looks good, thanks!

**Pull request merged**